src/Microsoft.SqlTools.ServiceLayer/Profiler/ProfilerSessionMonitor.cs

```diff
@@ -72,23 +73,21 @@ public void AddSessionListener(IProfilerSessionListener listener)
             }
         }
 
-        /// <summary>
-        /// Start monitoring the provided session
-        /// </summary>
-        public bool StartMonitoringSession(string viewerId, IXEventSession session)
+        public bool StartMonitoringStream(string viewerId, IXEventSession session, ConnectionInfo connInfo)
         {
             lock (this.sessionsLock)
             {
                 // start the monitoring thread
                 if (this.processorThread == null)
                 {
-                    this.processorThread = Task.Factory.StartNew(ProcessSessions);
+                    this.processorThread = Task.Factory.StartNew(ProcessStreams);
                 }
 
                 // create new profiling session if needed
                 if (!this.monitoredSessions.ContainsKey(session.Id))
                 {
                     var profilerSession = new ProfilerSession();
+                    profilerSession.ConnectionInfo = connInfo;
```

Comment:
if we really need the connection I would recommend storing it in the xeventsession interface, since the session is supposed to be connected already.